src/kani-compiler/rustc_codegen_kani/src/utils/utils.rs

```diff
@@ -16,6 +18,14 @@ pub fn dynamic_fat_ptr(typ: Type, data: Expr, vtable: Expr, symbol_table: &Symbo
     Expr::struct_expr(typ, btree_string_map![("data", data), ("vtable", vtable)], symbol_table)
 }
 
+pub fn emit_concurrency_warning(intrinsic: &str, loc: Location) {
```

Comment:
Does moving this here affect our ability to filter warnings? Might be better as a macro.

Reply:
It does not affect filtering, that depends on the `warn` macro. I was actually trying to avoid adding macros to the intrinsics work given the various readability/debugging issues they introduce, but if there is a good reason to have it as macro instead we can make the change.

Reply:
That is a valid concern and I agree with it, so the new version rewrites this as a macro.